debugging

```diff
@@ -0,0 +1,511 @@
+---
+layout: page
+title: Debugging Flutter Applications
+permalink: /debugging/
+---
+
+There are a wide variety of tools and features to help debug Flutter
+applications.
+
+## The Dart Analyzer
+
+Before running your applications, test your code with `flutter
+analyze`. This tool (which is a wrapper around the `dartanalyzer`
+tool) will analyze your code and help you find possible mistakes.
+
+The Dart analyzer makes heavy use of type annotations that you put in
+your code to help track problems down. You are encouraged to use them
+everywhere (avoiding `var`, untyped arguments, untyped list literals,
+etc) as this is the quickest and least paintful way of tracking down
+problems.
+
+## Dart Observatory (statement-level single-stepping debugger and profiler)
+
+If you started your application on an Android device using `flutter
+start`, then, while it is running, you can open the Web page at
+[http://localhost:8181/](http://localhost:8181/) to connect to your
+application directly with a statement-level single-stepping debugger.
+Observatory also supports profiling, examining the heap, etc.
```

Comment:
You should talk about running with `--no-checked` when profiling.  Otherwise you'll just be profiling all the asserts and debugging statements.